### DISCUSSION TITLE
Run Active File in Terminal - Mac Admin (sudo)

### DISCUSSION THREAD

**Discussion by JohnEMagee** [smartcontractkit/full-blockchain-solidity-course-py#1206] [Q&A]
The way my python and vs code set up is running, I can not use _run active file_ automatically.  The terminal error I receive is 
**zsh: permission denied: /Users/johnmagee/Documents/SolidityFCC/web3_py_simple_storage/deploy.py**
Obviously when I type _sudo python3 deploy.py_ it works fine but I have to do that manually.  Is there a way to set up VS Code so that it has the admin permission to run in the terminal?

Thank you

**Comment by cromewar** [comment]
Depends, I actually suggest to not use a file path with Admin privileges, but you can run as sudo vscode, and also set up it to run as admin with sudo as default.

**Reply by JohnEMagee** [reply]
I've tried to work around this a variety of ways but am not fluent enough in linux stuff to make it work, running it on admin on my home computer so far doesn't seem to be a dangerous issue...heck I can't even install many of the plugins properly without sudo (Mac m1)

**Reply by cromewar** [reply]
Well using sudo is a bad practice for security reasons, the best work around is to install the tools on paths which does not need sudo privilegies (not into /usr or /opt for example). However as I saw you also have those permissions on your home folder (documents) which is strange (may be M1 macs came with that feature). 
So you have some options.
1. keep using sudo each time (please be sure you know what you are installing).
2. change the folder privilegies on your home folder (this is a little bit more harder, as Mac is based on Unix [this](https://www.pluralsight.com/blog/it-ops/linux-file-permissions) guide should be helpful).

**Comment by JohnEMagee** [comment]
Unfortunately as I’ve done it admin style from the beginning of using python it’s just easy right now, if I reset my set up I’ll look into the other way thanks for the suggestions
